nodebuilder/share/opts.go

```diff
@@ -11,14 +13,22 @@ import (
 
 // WithPeerManagerMetrics is a utility function to turn on peer manager metrics and that is
 // expected to be "invoked" by the fx lifecycle.
-func WithPeerManagerMetrics(m *peers.Manager) error {
-	return m.WithMetrics()
+func WithPeerManagerMetrics(managers map[string]*peers.Manager) error {
```

Comment:
What is the reason to use map[string]*peers.Manager? Is it just the way to pass multiple components of same type?

Reply:
we provide this as a type to fx bc we need to pass in the managers in the correct order to shrex getter, so unless I provide them as an []*peers.Manager as well, this will have to do :/

Reply:
Thank you! It didn't clicked for me initially, but now I got the idea!